### PR TITLE
Reboot guest after create VFs

### PR DESCRIPTION
**Pull request by yanglei-rh**
Automation a new case to support reboot guest after create VFs

ID:2012
Signed-off-by: Lei Yang leiyang@redhat.com

### CODE REVIEW

**Comment by yanglei-rh** [comment]
Test pass:
(1/2) Host_RHEL.m9.u5.ovmf.qcow2.virtio_scsi.up.igb.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.igb_vf_reboot.q35: PASS (115.61 s

**Comment by yanglei-rh** [comment]
Hello @PaulYuuu Could you please help me review this patch? Many thanks.

File: provider/igb_create_vf.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+
+@error_context.context_aware
+def create_vf(test, params, session):
+    """
+    Create igb VFs devices
+    """
+    error_context.context("Create igb VFs devices", test.log.info)
+    pci_id = params.get("get_pci_id")
+    nic_pci = session.cmd_output(pci_id).strip().replace(":", r"\:")
```
[the remark this answers is not in the document]
Comment (PaulYuuu) [reply provider/igb_create_vf.py:11]: But I think you can use `utils_sriov.set_vf`

[the remark this answers is not in the document]
Comment (yanglei-rh) [reply provider/igb_create_vf.py:11]: > But I think you can use `utils_sriov.set_vf`

Thanks for the suggestion, I will change it in next update.

File: qemu/tests/cfg/igb_vf_reboot.cfg [hunk 2]
```diff
@@ -0,0 +1,9 @@
+- igb_vf_reboot:
+    virt_test_type = qemu
+    type = igb_vf_reboot
+    only igb
+    only Linux
+    reboot_method = shell
+    login_timeout = 240
+    get_pci_id = lspci -D |grep -i Eth |awk '{print $1}'
```
[the remark this answers is not in the document]
Comment (yanglei-rh) [reply qemu/tests/cfg/igb_vf_reboot.cfg:8]: > Ensure the guest only has 1 NIC.

Also will add this condition in the next update.

**Comment by yanglei-rh** [comment]
Hi @PaulYuuu Please help me review it again when you have free time, thanks a lot.

**Comment by yanglei-rh** [comment]
Test pass:
(1/1) Host_RHEL.m9.u5.ovmf.qcow2.virtio_scsi.up.igb.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.emulate_vf_reboot.q35: PASS (90.87 s)

**Comment by yanglei-rh** [comment]
Hi @PaulYuuu Please help me review this patch again, thanks a lot.

File: qemu/tests/emulate_vf_reboot.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
+from virttest import error_context
+
+from provider import create_emulate_vf
```
Comment (PaulYuuu) [inline qemu/tests/emulate_vf_reboot.py:3]: I am thinking about the module name, if it is only to create guest vfs, that's not enough to be a module. Also, the module name is `create_emulate_vf`, which means it only for creating VFs.

Should make sure it is only be used for VF? Or will add other functions to handle others in the future, like PF, or bind drivers, and then can define a suitable name.

**Comment by yanglei-rh** [comment]
@PaulYuuu Could you please help me review it again? Thanks a lot. It can test pass from my side.

File: qemu/tests/emulate_vf_reboot.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    emulate VFs reboot test:
+    1) Log into a guest
+    3) create max number vf for PF
+    4) Reboot guest
+    5) Log into the guest to verify it's up again
```
Comment (PaulYuuu) [inline qemu/tests/emulate_vf_reboot.py:12]: 1345?, index error?

**Review by PaulYuuu — COMMENTED** [review]
Others LGTM

File: qemu/tests/emulate_vf_reboot.py (hunk 4 above)
Comment (yanglei-rh) [reply qemu/tests/emulate_vf_reboot.py]: My mistake, updated.

**Comment by yanglei-rh** [comment]
Hi @PaulYuuu Please help me review it again, thanks a lot.

**Review by PaulYuuu — APPROVED** [review]

**Pull request merged**